fvtest/jitbuildertest/CallReturnTest.cpp

```diff
@@ -0,0 +1,173 @@
+   return val;
+   }
+
+static int8_t
+testInt8(int8_t val)
+   {
+   #define TEST_INT8_LINE LINETOSTR(__LINE__)
+   return val;
+   }
+
+DEFINE_BUILDER(TestInt64ReturnType,
+               Int64,
+               PARAM("param", Int64))
+   {
+   DefineFunction((char *)"testInt64",
+                  (char *)__FILE__,
+                  (char *)TEST_INT64_LINE,
+                  (void *)&testInt64,
+                  Int64,
+                  1,
+                  Int64);
+
+   TR::IlValue *param = Load("param");
+   TR::IlValue *int64Result = Call("testInt64", 1, param);
+   TR::IlValue *int64Val = ConstInt64(1);
+   TR::IlValue *result = Add(int64Result, int64Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt32ReturnType,
+               Int32,
+               PARAM("param", Int32))
+   {
+   DefineFunction((char *)"testInt32",
+                  (char *)__FILE__,
+                  (char *)TEST_INT32_LINE,
+                  (void *)&testInt32,
+                  Int32,
+                  1,
+                  Int32);
+
+   TR::IlValue *param = Load("param");
+   TR::IlValue *int32Result = Call("testInt32", 1, param);
+   TR::IlValue *int32Val = ConstInt32(1);
+   TR::IlValue *result = Add(int32Result, int32Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt16ReturnType,
+               Int16,
+               PARAM("param", Int16))
+   {
+   DefineFunction((char *)"testInt16",
+                  (char *)__FILE__,
+                  (char *)TEST_INT16_LINE,
+                  (void *)&testInt16,
+                  Int16,
+                  1,
+                  Int16);
+
+   TR::IlValue *param = Load("param");
+   TR::IlValue *int16Result = Call("testInt16", 1, param);
+   TR::IlValue *int16Val = ConstInt16(1);
+   TR::IlValue *result = Add(int16Result, int16Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt8ReturnType,
+               Int8,
+               PARAM("param", Int8))
+   {
+   DefineFunction((char *)"testInt8",
+                  (char *)__FILE__,
+                  (char *)TEST_INT8_LINE,
+                  (void *)&testInt8,
+                  Int8,
+                  1,
+                  Int8);
+
+   TR::IlValue *param = Load("param");
+   TR::IlValue *int8Result = Call("testInt8", 1, param);
+   TR::IlValue *int8Val = ConstInt8(1);
+   TR::IlValue *result = Add(int8Result, int8Val);
+
+   Return(result);
+
+   return true;
+   }
+
+class ReturnTypeTest : public JitBuilderTest {};
+
+typedef int64_t (*Int64ReturnType)(int64_t);
+TEST_F(ReturnTypeTest, Int64_Test)
+   {
+   Int64ReturnType testFunction;
+   ASSERT_COMPILE(TR::TypeDictionary, TestInt64ReturnType, testFunction);
+   ASSERT_EQ(testFunction(1), 2);
+   }
+
+typedef int32_t (*Int32ReturnType)(int32_t);
+TEST_F(ReturnTypeTest, Int32_Test)
+   {
+   Int32ReturnType testFunction;
+   ASSERT_COMPILE(TR::TypeDictionary, TestInt32ReturnType, testFunction);
+   ASSERT_EQ(testFunction(1), 2);
```

Comment:
Maybe use an input value that requires more than 16 bits, just for completeness :) .